ngraph/core/src/op/gru_cell.cpp

```diff
@@ -104,151 +107,102 @@ bool op::v3::GRUCell::visit_attributes(AttributeVisitor& visitor)
     return op::util::RNNCellBase::visit_attributes(visitor);
 }
 
-void op::v3::GRUCell::pre_validate_and_infer_types()
+void op::v3::GRUCell::validate_and_infer_types()
 {
-    set_output_type(0, get_input_element_type(0), PartialShape::dynamic());
+    std::vector<ngraph::PartialShape> input_param{};
```

Comment:
You use `input_param` only to pas it to the `validate_input_rank_dimension` function. Can you instead use  `x_pshape`, `ht_pshape`, etc which you get in lines 125-129?